SU2_CFD/src/numerics_direct_turbulent.cpp

```diff
@@ -1272,10 +1245,10 @@ void CSourcePieceWise_TurbSST::ComputeResidual(su2double *val_residual, su2doubl
 
    /*--- Implicit part ---*/
 
-   val_Jacobian_i[0][0] = -beta_star*TurbVar_i[1]*Volume;
-   val_Jacobian_i[0][1] = -beta_star*TurbVar_i[0]*Volume;
+   val_Jacobian_i[0][0] = -beta_star*Density_i*TurbVar_i[1]*Volume;
+   val_Jacobian_i[0][1] = -beta_star*Density_i*TurbVar_i[0]*Volume;
    val_Jacobian_i[1][0] = 0.0;
-   val_Jacobian_i[1][1] = -2.0*beta_blended*TurbVar_i[1]*Volume;
+   val_Jacobian_i[1][1] = -2.0*beta_blended*Density_i*TurbVar_i[1]*Volume;
   }
```

Comment:
Damn, I've been staring at these lines of code for a while and I never realized that Density was missing from the Jacobian computation. Good catch!

Reply:
It was not missing, before the Jacobians were w.r.t. rho*k and rho*w, that is why in the diffusion part they had to be divided by density, I changed it to k and w (only for the SST model) since the mass flux already contains rho. I am not 100% sure this is correct though... In fact CTurbSolver updates the solution via `node[iPoint]->AddConservativeSolution`. So a better catch by you actually!